src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -73,6 +76,12 @@ internal class FhirCosmosSearchService : SearchService
             _logger = logger;
             _resourceTypeSearchParameter = SearchParameterInfo.ResourceTypeSearchParameter;
             _resourceIdSearchParameter = new SearchParameterInfo(SearchParameterNames.Id, SearchParameterNames.Id);
+
+            _expressionRewriters = new Lazy<IReadOnlyCollection<IExpressionVisitorWithInitialContext<object, Expression>>>(() => expressionRewriters
+                .OrderBy(x => x.Order)
+                .Cast<IExpressionVisitorWithInitialContext<object, Expression>>()
+                .Append(DateTimeEqualityRewriter.Instance)
```

Comment:
Do we have tests which combine compartment search with DateTime comparison? Is that even possible?

Reply:
There shouldn't be any restriction on multiple rewriters being used. I thought I there was a case in the Patient Everything tests, but I wasn't able to get both rewriters to fire